core/arch/arm/plat-nuvoton/main.c

```diff
@@ -0,0 +1,77 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2016-2023, Linaro Limited
+ * Copyright (c) 2014-2023, STMicroelectronics International N.V.
+ * Copyright (C) 2022-2023 Nuvoton Ltd.
+ */
+
+#include <console.h>
+#include <drivers/gic.h>
+#include <drivers/ns16550.h>
+#include <kernel/boot.h>
+
+#if defined(CFG_NPCM_DEBUG)
+#include <kernel/linker.h>
+#include <trace.h>
+#endif /* CFG_NPCM_DEBUG */
+
+#include <platform_config.h>
+
+#if defined(CFG_NPCM_DEBUG)
+#define COLOR_NORMAL	"\x1B[0m"
+#define COLOR_RED	"\x1B[31m"
+#define COLOR_GREEN	"\x1B[32m"
+#define COLOR_YELLOW	"\x1B[33m"
+#define COLOR_BLUE	"\x1B[34m"
+#define COLOR_MAGENTA	"\x1B[35m"
+#define COLOR_CYAN	"\x1B[36m"
+#define COLOR_WHITE	"\x1B[37m"
+#endif /* CFG_NPCM_DEBUG */
+
+static struct gic_data gic_data __nex_bss;
+static struct ns16550_data console_data __nex_bss;
+
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE, UART_REG_SIZE);
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GICD_BASE, GIC_DIST_REG_SIZE);
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GICC_BASE, GIC_DIST_REG_SIZE);
+
+register_ddr(DRAM0_BASE, DRAM0_SIZE);
+
+#if defined(CFG_NPCM_DEBUG)
```

Comment:
prefer removing this condition and use:
```c
	if (IS_ENABLED(CFG_NPCM_DEBUG))
		print_version();
```
when called. It allows build to check syntax even when the switch is not enabled.
By the way, to get this trace as early as possible, I would recommend to call `print_version()` from `console_init()` (at the end of the function) rather than from `main_init_gic()`. but that is not mandatory. Do as you prefer.

Reply:
Compilation error: implicit declaration of function 'print_version' [-Werror=implicit-function-declaration]
51 | print_version();